proposals/NNNN-android-transport-overhaul.md

```diff
@@ -0,0 +1,491 @@
+
+<center>**------- Will be updated ------**</center>
+
+![Old Layers](../assets/proposals/NNNN-android-transport-overhaul/TransportOverhaul_old.png "Old transport layers")
+
+The first implementation will look like this for a minor change.
+![Bridge Layers](../assets/proposals/NNNN-android-transport-overhaul/TransportOverhaul_bridge.png "Bridged transport layers")
+
+
+After the refactor the layers will look like this.
+
+![New Layers](../assets/proposals/NNNN-android-transport-overhaul/TransportOverhaul_new.png "New transport layers")
+
+#### Move the protocol layer 
+
+The protocol layer needs to be moved away from inside the transport layer. It actually needs to be the exact opposite. The protocol layer should sit ontop of the transport layer. A new protocol layer will be created, `SdlProtocol` that will be instantiated into the `SdlSession` class. All control frame messages and related `SdlPacket` messages will be sent into the `WiProProtocl` instance for handling from the transport layer. The protocol layer will then determine whether or not packets need to move up the stack. 
+
+#### Transport Manager
+
+This will be the new bridge between the transports and the `SdlRouterService`. All the data gathered and delivered to the library will be passed to the `SdlProtocol`. This way, the `SdlProtocol` can decide where packets should go in logical blocks instead of the transport layer.
+
+#### All Transports to Router Service
+
+The next piece required for this refactor will be to move all the transport connections into the `SdlRouterService`. This is a logical step for AOA as discussed earlier. At that point 2 out of 3 transports would be already located in the router service. This means transport switching and the ability to use primary and secondary transports will already need to exist in the router service. Therefore, we should remove redundant logic and reduce the scope of transport classes outside of the router service. The only way this can be accomplished is by also moving the TCP transport.
+
+#### ForceOnConnect Removal
+
+The `ForceOnConnect` extra will no longer be used so the logic for it can be removed. This piece of integration has always been extra clutter and very error prone. It's removal from necessity will make the library more stable and less developer error prone.
+
+#### AOA and TCP Multiplexing
+
+A new USB and TCP transport discussed later, will be implemented into the `SdlRouterService`. This will allow all apps to connect through the USB and TCP transports when it is available. 
+
+###### TCP Single SDL Port Assumption
+A current assumption about the TCP transport is that the SDL core module will only expose a single port for all apps to connect through for a secondary transport. 
+
+The ability will be added to modify this in the future so that the router service can hold multiple transports of the same type, but for unless it is required the feature can be implemented at a latter time. 
```

Comment:
Please update the last part of this sentence ("but for unless it is required...") and change "latter time." to "later time."